src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2482,7 +2482,7 @@ private SyntaxListBuilder LexConflictMarkerDisabledText(ref SyntaxListBuilder tr
             while (true)
             {
                 var ch = this.TextWindow.PeekChar();
-                if (ch == SlidingTextWindow.InvalidCharacter)
+                if (ch == SlidingTextWindow.InvalidCharacter && TextWindow.IsReallyAtEnd())
```

Comment:
How many PRs should I create? I wrote in the original post that different commits do different things and it's best to review commit by commit.

Reply:
> How many PRs should I create?

Either 2 or 3.  One PR for perf, and then either 1 or 2 prs for the fixes (i'm ok with teh fixes going in on one pr).

> I wrote in the original post that different commits do different things and it's best to review commit by commit.

Compiler PRs are squashed.  So it will all be one commit when it goes in.  As separate PRs though we can keep the changes cleanly separated.

Reply:
Removed.